lib/pdk/util/bundler.rb

```diff
@@ -92,18 +93,32 @@ def locked?
           !gemfile_lock.nil? && File.file?(gemfile_lock)
         end
 
-        def installed?(gem_overrides = {})
-          PDK.logger.debug(_('Checking for missing Gemfile dependencies.'))
-
-          argv = ['check', "--gemfile=#{gemfile}", '--dry-run']
-
-          cmd = bundle_command(*argv).tap do |c|
-            c.update_environment(gemfile_env(gem_overrides)) unless gem_overrides.empty?
+        # @return [Bundler::Definition]
+        # @param gem_overrides [Hash] one of three keys that contain puppet version
+        def gemfile_definition(gem_overrides = {})
+          gem_overrides ||= {}
+          @gemfile_definition ||= begin
+            env_vars = gemfile_env(gem_overrides)
+            env_vars.each { |name, value| ENV[name.to_s] = value }
+            ::Bundler.configure
+            definition = ::Bundler::Definition.build(gemfile, gemfile_lock, nil)
+            # reset env variable back to nil so we don't cause environment polution
+            env_vars.each { |name, _value| ENV[name.to_s] = nil }
+            definition
           end
+        end
 
-          result = cmd.execute!
+        # @return [Boolean] - true if the gemfile with the given overrides is installed
+        # @param gem_overrides [Hash] one of three keys that contain puppet version
+        # @example installed?({puppet: '6.10.0'})
+        # @note does not allow you to check arbitrary gems not already in the gemfile
+        def installed?(gem_overrides = {})
+          return false if !File.file?(gemfile) && ::Bundler.frozen_bundle?
 
-          result[:exit_code].zero?
+          definition = gemfile_definition(gem_overrides)
+          definition.validate_runtime!
+          # determines if something is not installed
+          !definition.missing_specs?
```

Comment:
Perhaps log at `debug` level the gems that were missing? You can currently see that at `debug` in the raw `bundle` output.